src/tls/incoming.rs

```diff
@@ -255,14 +271,39 @@ impl AsyncWrite for MaybeTlsIncomingStream<TcpStream> {
     }
 
     fn poll_shutdown(self: Pin<&mut Self>, cx: &mut Context) -> Poll<io::Result<()>> {
-        self.poll_io(cx, |s, cx| s.poll_shutdown(cx))
-    }
-
-    fn poll_write_buf<B: Buf>(
-        self: Pin<&mut Self>,
-        cx: &mut Context,
-        buf: &mut B,
-    ) -> Poll<io::Result<usize>> {
-        self.poll_io(cx, |s, cx| s.poll_write_buf(cx, buf))
+        let mut this = self.get_mut();
+        match &mut this.state {
+            StreamState::Accepted(stream) => {
+                let pinned = Pin::new(&mut *stream);
+                match pinned.poll_flush(cx) {
```

Comment:
Do we need this flush, since the following `poll_shutdown` implies flushing?

Reply:
Good catch!

Reply:
Done with efac5c3.